app/Http/Controllers/Api/MasjidProfileController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+use Facades\App\Helpers\Setting;
+
+class MasjidProfileController extends Controller
+{
+    public function updateLogo(Request $request)
+    {
+        $validatedPayload = $request->validate([
+            'image' => 'required'
+        ]);
+
+        if (!base64_decode($validatedPayload['image'])) {
+            return response()->json([
+                'message' => __('masjid_profile.image_not_found'),
+            ]);
+        }
+
+        if ($masjidLogoPath = Setting::get('masjid_logo_path')) {
+            Storage::delete($masjidLogoPath);
+        }
+
+        $imageParts = explode(';base64,', $validatedPayload['image']);
+        $imageBase64 = base64_decode($imageParts[1]);
+        $imageName = uniqid() . '.webp';
+
+        Storage::put($imageName, $imageBase64);
+        Setting::set('masjid_logo_path', $imageName);
+
+        return response()->json([
+            'message' => __('masjid_profile.logo_uploaded'),
+            'image' => asset($imageName)
```

Comment:
Bagian sini mesti pakai Storage::url agar bisa tampil sesuai URL yang dari storage.
```suggestion
            'image' => Storage::url($imageName),
```